### ISSUE TITLE
[Feature Request] Add compatibility with .eml and .msg files

### ISSUE THREAD

**Issue by aidenmitchell** [freedomofpress/dangerzone#163]
It would be nice to have the ability to convert .eml (email) files, as those can contain malicious link/images/anything you name.

**Comment by andrewcrook** [comment]
I think thanks a whole project in itself

**Comment by deeplow** [comment]
Large document leaks / dumps include `.eml`. This may be worth exploring as with these file types we can probably print them as individual PDFs. `.inbox` files would be a bit more complicated, though since one file can lead to many other.

**Comment by apyrgio** [comment]
This need has resurfaced: https://github.com/freedomofpress/dangerzone/issues/760

**Comment by sea-kelp** [comment]
Hi @apyrgio, I see you mentioned in https://github.com/freedomofpress/dangerzone/issues/760 that eml/msg support was blocked by issues in the 0.7.0 release. Now that 0.7.0 has been released, would it be possible to restart discussions around adding this feature?

**Comment by mozillamonks** [comment]
Adding in support for this as well; a lot of the FOIA responses I process through Muckrock come as .msg files which often have untrusted remote images and attachments as well. Even just getting the text cleanly out of these would be incredibly handy.

**Comment by apyrgio** [comment]
I can't promise much, but this is something I'd like to tackle for our 0.11.0 release. Implementation-wise, I have the following in mind that we need to figure out first.

## Attachments

Like `.zip` files, `.eml` / `.msg` files include other files within them. How should Dangerzone extract/present these embedded files in a safe and user-friendly way?

The simplest way to deal with them is like a PDF document. We could render first the body of the email, and then start rendering the attachments as subsequent pages, probably by prefacing that this is an attachment. A more complex way is to create a directory with PDFs, one PDF for the email body, and one PDF per attachment. The problem here is that you can't "render" a directory. We'd need to change Dangerzone's app <-> container protocol from:

```
<number of pages> [uint16]
  <page width> [uint16]
  <page height> [uint16]
  <page data> [varchar]
...
```

(presumably) to:

```
<number of documents> [uint16]
  <document name length> [uint16]
  <document name> [varchar]
  <number of pages> [uint16]
   <page width> [uint16]
   <page height> [uint16]
   <page data> [varchar]
...
```

And a few more considerations:
1. We'd need to make sure that an attacker-controlled name cannot pose more danger to the user than it does so already. If we nail this, then we can possibly nail `.zip` / `.tar` files as well.
2. How to handle files that Dangerzone can't or fails to render?
3. We should never render the .eml file as text, since the attachments can be fully restored from their base-64 representation, and thus their metadata.

## Rendering emails

1. Emails may contain an HTML body, or a text one, or both. What do we render in this case? How do we explain to the user the difference?
2. HTML bodies may contain active components. We need to strip those before we render them.
3. The solution needs to be small enough, so that it doesn't bloat our already large container image. Also, it should work on Qubes, i.e., the dependencies must be installed as a Fedora package.

## Metadata

One of Dangerzone's goal is to render just the document content and remove the metadata, in case a source wants to share a sanitized version with a third-party. What type of metadata should we remove here? DKIM signatures and connection logs are probably the first ones to go, but what about `bcc:` fields, dates, or even `to: / cc:` fields?

Also, at least until now, the files that Dangerzone sanitized looked like the original file. In this case, if we decide to include everything into a single PDF, it would be easy to figure out that the sender has sanitized the an email with Dangerzone.
